lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -15,6 +15,74 @@ local previewers = {}
 
 local ns_previewer = vim.api.nvim_create_namespace('telescope.previewers')
 
+local color_hash = {
+  ['d'] = 'Directory',
+  ['.'] = 'Normal',
+  ['r'] = 'Constant',
+  ['w'] = 'Statement',
+  ['x'] = 'String',
+  ['-'] = 'Comment',
+  [2] = 'String',      -- SIZE
+  [3] = 'Constant',    -- USER
+  [4] = 'Constant',    -- GROUP
+  [5] = 'Directory',   -- DAY OF MONTH
+  [6] = 'Directory',   -- MONTH
+  [7] = 'Directory',   -- YEAR or TIME
+  [8] = function(line) -- DIR or FILE
+    if line:sub(1, 1) == 'd' then
+      return 'Directory'
+    else
+      return 'Normal'
+    end
+  end,
+}
+
+-- TODO(conni2461): Is there a better way to do this?
+local get_sections = function(line)
+  local sections = {}
+  local current_str = ''
+  local start_section = 0
+  for i = 1, #line do
+    local c = line:sub(i, i)
+    if start_section == 0 then start_section = i end
+    if c == ' ' then
+      if current_str ~= '' then
+        table.insert(sections, { str = current_str, start_index = start_section, end_index = i })
+        current_str = ''
+      end
+      start_section = 0
+    elseif i == #line then
+      current_str = current_str .. c
+      table.insert(sections, { str = current_str, start_index = start_section, end_index = i + 1 })
+    else
+      current_str = current_str .. c
+    end
+  end
+  return sections
+end
+
+local colorize_ls = function(bufnr, data)
+  for lnum, line in ipairs(data) do
+    local sections = get_sections(line)
+    for i = 1, #sections[1].str do -- Highlight permissions
+      local c = sections[1].str:sub(i, i)
+      vim.api.nvim_buf_add_highlight(bufnr, ns_previewer, color_hash[c], lnum - 1, i - 1, i)
+    end
+    -- TODO(conni2461): Change hardcoded colors
+    for i = 2, #sections do
+      local color = color_hash[i]
+      color = type(color) == 'function' and color(line) or color
+      vim.api.nvim_buf_add_highlight(bufnr,
```

Comment:
I managed to break this line, let me find a way to reproduce it.

Reply:
ok I found it. If we have a directory with a space in its name, it breaks. I was wildly guessing (without reading the code) that you use a space for splitting the section so I decided to try it out.

Reply:
Yeah doing that. I have thought about having spaces in filenames and then i forgot about it 🤣 One sec

Reply:
Btw thanks good catch

Reply:
Should work now.

Reply:
Oh windows might be broken 😆 They get a different output. I do that later